app.ts

```diff
@@ -0,0 +1,310 @@
+const IMQS = "http://localhost:2050";
+let totalRecordCount: number;
+let totalPages: number;
+let currentValueOfFirstRecord = 1;
+let currentFirstRecordIndex = 1;
+let currentPage: number;
+let recordsPerPage: number;
+let searchedIndex: number | null = null;
+let resizeTimeout: number;
+
+
+async function totalRecords(): Promise<void> {
+	try {
+		let recordCountResponse = await fetch(`${IMQS}/recordCount`);
+		let recordCountText = await recordCountResponse.text();
+		let recordCount = parseInt(recordCountText);
+		totalRecordCount = recordCount;
+	} catch (error) {
+		throw (error)
+	};
+};
+async function displayColumns() {
+	try {
+		let columnsResponse = await fetch(`${IMQS}/columns`);
+		let columns = await columnsResponse.json();
+		let tableHeaderRow = $("#tableHeaderRow");
+		columns.forEach((columnName: string) => {
+			let th = document.createElement("th");
+			th.textContent = columnName;
+			tableHeaderRow.append(th);
+		});
+		await totalRecords();
+		totalPages = Math.ceil(totalRecordCount / recordsPerPage);
+		await updatePages();
+	} catch (error) {
+		throw (error)
+	};
+};
+async function displayData(fromRecord: number, recordsDisplayed: number) {
+	let isSearchMode = false;
+	let searchResultIndexes: number[] = [];
+	try {
+		fromRecord = Math.max(fromRecord, 0);
+		const maximumRecords = Math.min(
+			totalRecordCount - fromRecord,
+			recordsDisplayed
+		);
+		let response = await fetch(
+			`${IMQS}/records?from=${fromRecord}&to=${fromRecord + maximumRecords - 1
+			}`
+		);
+		let dataText = await response.text();
+		let data = JSON.parse(dataText);
+		let tableData = "";
+		if (currentFirstRecordIndex <= 1) {
+			$("#prevPageButton").hide();
+		} else {
+			$("#prevPageButton").show();
+		};
+		const lastRecordOnPage = currentFirstRecordIndex + recordsPerPage - 1;
+		if ((lastRecordOnPage === totalRecordCount - 1 || searchedIndex !== null) && lastRecordOnPage === totalRecordCount - 1) {
+			$("#nextPageButton").hide();
+		} else {
+			$("#nextPageButton").show();
+		};
+		let isSearchMode = false;
+		let searchResultIndexes: number[] = [];
+		let searchResultDisplay = false;
+		const searchIndexOnPage = searchResultIndexes[0] - fromRecord;
+		if (searchResultDisplay) {
+			const searchIndex = searchResultIndexes[0];
+			const searchPage = Math.ceil((searchIndex + 1) / recordsPerPage);
+			const newFromRecord = (searchPage - 1) * recordsPerPage;
+			const newToRecord = newFromRecord + recordsPerPage - 1;
+			const actualFromRecord = Math.min(newFromRecord + 1, totalRecordCount);
+			const actualToRecord = Math.min(newToRecord + 1, totalRecordCount);
+			fromRecord = newFromRecord;
+			recordsDisplayed = recordsPerPage;
+			searchResultIndexes.forEach((searchIndex: any) => {
+				const rowIndex = searchIndex % recordsPerPage;
+				const rowElement = $("#tableBody tr").eq(rowIndex);
+				rowElement.css("background-color", "var(--results-color)");
+			});
+			searchResultDisplay = false;
+		}
+		else if (!isSearchMode) {
```

Comment:
Will always go in here as you declare it as false right before the if statement. `else` should be on the same line as the `}`